src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedNewTests.cs

```diff
@@ -0,0 +1,928 @@
+                //         var v1 = (Struct)new();
+                Diagnostic(ErrorCode.ERR_DefaultValueTypeCtorInTargetTypedNew, "new()").WithArguments("Struct").WithLocation(12, 26),
+                // (13,18): error CS9366: The type 'Action' may not be used as the target-type of 'new'.
+                //         var v2 = (Action)new();
+                Diagnostic(ErrorCode.ERR_BadTargetTypeForNew, "(Action)new()").WithArguments("System.Action").WithLocation(13, 18),
+                // (14,18): error CS0716: Cannot convert to static type 'Static'
+                //         var v3 = (Static)new();
+                Diagnostic(ErrorCode.ERR_ConvertToStaticClass, "(Static)new()").WithArguments("Static").WithLocation(14, 18),
+                // (14,9): error CS0723: Cannot declare a variable of static type 'Static'
+                //         var v3 = (Static)new();
+                Diagnostic(ErrorCode.ERR_VarDeclIsStaticClass, "var").WithArguments("Static").WithLocation(14, 9),
+                // (15,28): error CS0144: Cannot create an instance of the abstract class or interface 'Abstract'
+                //         var v4 = (Abstract)new();
+                Diagnostic(ErrorCode.ERR_NoNewAbstract, "new()").WithArguments("Abstract").WithLocation(15, 28),
+                // (16,18): error CS9366: The type 'Interface' may not be used as the target-type of 'new'.
+                //         var v5 = (Interface)new();
+                Diagnostic(ErrorCode.ERR_BadTargetTypeForNew, "(Interface)new()").WithArguments("Interface").WithLocation(16, 18),
+                // (17,18): error CS9366: The type 'Enumeration' may not be used as the target-type of 'new'.
+                //         var v6 = (Enumeration)new();
+                Diagnostic(ErrorCode.ERR_BadTargetTypeForNew, "(Enumeration)new()").WithArguments("Enumeration").WithLocation(17, 18),
+                // (18,23): error CS9367: The default constructor of the value type 'int' may not be used with target-typed 'new'. Consider using 'default' instead.
+                //         var v7 = (int)new();
+                Diagnostic(ErrorCode.ERR_DefaultValueTypeCtorInTargetTypedNew, "new()").WithArguments("int").WithLocation(18, 23),
+                // (19,18): error CS1919: Unsafe type 'int*' cannot be used in object creation
+                //         var v8 = (int*)new();
+                Diagnostic(ErrorCode.ERR_UnsafeTypeInObjectCreation, "(int*)new()").WithArguments("int*").WithLocation(19, 18),
+                // (20,24): error CS9367: The default constructor of the value type 'int?' may not be used with target-typed 'new'. Consider using 'default' instead.
+                //         var v9 = (int?)new();
+                Diagnostic(ErrorCode.ERR_DefaultValueTypeCtorInTargetTypedNew, "new()").WithArguments("int").WithLocation(20, 24),
+                // (21,19): error CS8181: 'new' cannot be used with tuple type. Use a tuple literal expression instead.
```

Comment:
>Use a tuple literal expression instead. [](start = 80, length = 39)

Nit: Maybe the diagnostic should say "... or default literal ..."

Reply:
We report the same diagnostic for usage of the non-default ctor as well e.g. `(int, int) t = new(1, 2);`.

Note: I reused an existing error that we produce for `new (int, int)()` in 7.3.